src/components/DonationItemsTable/DonationItemsTable.jsx

```diff
@@ -44,21 +49,62 @@ const DonationItemsTable = () => {
 const DonationItems = ({ category }) => {
   const [data, setData] = useState([]);
   const { backend } = useBackend();
+  const [currentItemNum, setCurrentItemNum] = useState(0);
+  const [currentPageNum, setCurrentPageNum] = useState(1);
+  const [pageLimit, setPageLimit] = useState(1);
+  const [selectedItem, setSelectedItem] = useState(null);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
   const TABLE_HEADERS = ['Id', 'Name', 'Quantity Type', 'Price', 'Category'];
+
+  // useEffect(() => {
+  //   const getData = async () => {
+  //     try {
+  //       const response = await backend.get(`/value/filter/${category}`);
+  //       setData(response.data);
+  //       console.log(response.data);
+  //     } catch (error) {
+  //       console.error('Error getting donation items:', error);
+  //       setData([]);
+  //     }
+  //   };
+  //   getData();
+  // }, [backend]);
+
   useEffect(() => {
-    const getData = async () => {
+    const loadData = async () => {
       try {
-        const response = await backend.get(`/value/filter/${category}`);
-        setData(response.data);
+        loadInfo();
       } catch (error) {
-        console.error('Error getting donation items:', error);
-        setData([]);
+        console.error('Error fetching data:', error);
       }
     };
-    getData();
-  }, [backend, category]);
+    loadData();
+  }, [backend, category, currentPageNum, data]);
+
+  const loadInfo = async () => {
+    changePage();
+    const itemsNumResponse = await backend.get(`/value/totalValues`);
+    setCurrentItemNum(itemsNumResponse.data[0]['count']);
+    setPageLimit(Math.ceil(itemsNumResponse.data[0]['count'] / 10));
+  };
+
+  const changePage = async () => {
+    const formResponse = await backend.get(
+      `/value/?itemsLimit=10&pageNum=${currentPageNum}`,
+    );
+    setData(formResponse.data);
+  };
```

Comment:
make sure to pass the category to the route (a comment is left on the backend to change the route accordingly) so that it only grabs the corresponding data for the filter